includes/menu.inc

```diff
@@ -18,36 +18,33 @@ function oe_bootstrap_theme_preprocess_menu(&$variables, $hook) {
   $items = $variables['items'];
   foreach ($items as $key => $item) {
     $class = $item['in_active_trail'] ? ['active'] : [];
-    $menu_item = [
-      'label' => $item['title'],
-      'path' => $item['url']->toString(),
-      'attributes' => new Attribute(['class' => $class]),
-    ];
-    $variables['items'][$key] = $menu_item;
+    $variables['items'][$key]['label'] = $item['title'];
+    $variables['items'][$key]['path'] = $item['url']->toString();
+    $variables['items'][$key]['attributes'] = new Attribute(['class' => $class]);
 
     if (empty($item['below'])) {
       continue;
     }
 
     $class[] = 'nav-link';
-    $menu_item['attributes'] = new Attribute(['class' => $class]);
     $item_with_submenu = [
       'standalone' => TRUE,
       'dropdown' => TRUE,
       'link' => TRUE,
       'alignment' => 'dropdown-menu-sm-start',
-      'trigger' => $menu_item,
+      'trigger' => [
+        'label' => $item['title'],
+        'path' => $item['url']->toString(),
+        'attributes' => new Attribute(['class' => $class]),
+      ],
     ];
-    $variables['items'][$key] = $item_with_submenu;
+    $variables['items'][$key] += $item_with_submenu;
```

Comment:
@escuriola 
Overall I feel we have a lot of redundant juggling of variables here:
We are having local variables like `$item` from the foreach, `$items` from outside the foreach, `$item_with_submenu`, and in the old code we had `$menu_item`. But then we are writing each value to `$variables['items'][$key]`.

We have two options that would simplify this:
- By-reference `foreach ($variables['items'] as $key => &$item) {`, then write everything directly to `$item`. Perhaps we could call it `$item_ref` to remind the reader that this is by-reference.
- Don't use by-reference, write everything to `$item`, and in the end set `$variables['items'][$key] = $item;`.

Also, instead of local vars like `$item_with_submenu` with later `+=`, we could directly call `$item += [...]`.

----

I have a more general concern with the entire rewriting of variables here, but I will address this in another comment.